jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/MavenProjectProperties.java

```diff
@@ -211,30 +211,44 @@ public JibContainerBuilder createContainerBuilder(
         return JavaContainerBuilderHelper.fromExplodedWar(javaContainerBuilder, explodedWarPath);
       }
 
-      if (containerizingMode == ContainerizingMode.PACKAGED) {
-        String jarName = project.getBuild().getFinalName() + "." + project.getPackaging();
-        Path jar = Paths.get(project.getBuild().getDirectory(), jarName);
-        return javaContainerBuilder
-            .addDependencies(getDependencies())
-            .addToClasspath(jar)
-            .toContainerBuilder();
+      // Add dependencies
+      Set<Artifact> artifacts = project.getArtifacts();
+      List<Path> dependencies =
+          artifacts.stream().map(Artifact::getFile).map(File::toPath).collect(Collectors.toList());
+      javaContainerBuilder.addDependencies(dependencies);
+
+      switch (containerizingMode) {
+        case EXPLODED:
+          // Add resources, and classes
+          Path classesOutputDirectory = Paths.get(project.getBuild().getOutputDirectory());
+          Predicate<Path> isClassFile = path -> path.getFileName().toString().endsWith(".class");
```

Comment:
Just use `path.endsWith(".class")`?

Reply:
`Path.endsWith()` works on path elements, not on the entire string.

Reply:
Ah right.  I think I've asked this before.  Might be worth adding a comment.

Reply:
Comments added.